packages/core/src/components/fields/SchemaField.js

```diff
@@ -241,7 +242,25 @@ function SchemaFieldRender(props) {
   const FieldTemplate =
     uiSchema["ui:FieldTemplate"] || registry.FieldTemplate || DefaultTemplate;
   let idSchema = props.idSchema;
-  const schema = retrieveSchema(props.schema, rootSchema, formData);
+  var schema = retrieveSchema(props.schema, rootSchema, formData);
+
+  if (schema.allOf) {
+    for (var i = 0; i < schema.allOf.length; i++) {
+      let allOfSchema = schema.allOf[i];
+
+      // if we see an if in our all of schema then evaluate the if schema and select the then / else, not sure if we should still merge without our if then else
+      if ("if" in allOfSchema) {
+        allOfSchema = isValidAddDefaults(allOfSchema.if, formData)
+          ? allOfSchema.then
+          : allOfSchema.else;
+      }
+
+      if (allOfSchema) {
+        schema = mergeSchemas(schema, allOfSchema);
```

Comment:
@epicfaace please note that this merge is still incorrect as i believe the schema will take the last occurrence of an array, i.e. your example will take the last type, this is still incorrect as per https://json-schema.org/understanding-json-schema/reference/combining.html but this PR brings the allof functionality much closer to spec (in my mind)